protocols/imap/src/main/java/org/apache/james/imap/processor/AuthenticateProcessor.java

```diff
@@ -176,24 +175,25 @@ protected Closeable addContextToMDC(AuthenticateRequest request) {
     }
 
     private void doOAuth(String initialResponse, ImapSession session, ImapRequest request, Responder responder) {
-        if (session.oidcSaslConfiguration().isEmpty()) {
+        if (!session.isSupportOAuth()) {
             no(request, responder, HumanReadableText.UNSUPPORTED_AUTHENTICATION_MECHANISM);
         } else {
             OIDCSASLParser.parse(initialResponse)
-                .flatMap(oidcInitialResponseValue -> {
-                    OidcSASLConfiguration configuration = session.oidcSaslConfiguration().get();
-                    return new OidcJwtTokenVerifier().verifyAndExtractClaim(oidcInitialResponseValue.getToken(), configuration.getJwksURL(), configuration.getClaim());
-                })
+                .flatMap(oidcInitialResponseValue -> session.oidcSaslConfiguration()
+                    .flatMap(configuration -> new OidcJwtTokenVerifier().verifyAndExtractClaim(oidcInitialResponseValue.getToken(), configuration.getJwksURL(), configuration.getClaim())))
                 .flatMap(this::extractUserFromClaim)
-                .ifPresentOrElse(username -> {
-                    session.authenticated();
-                    session.setMailboxSession(getMailboxManager().createSystemSession(username));
-                    okSuccess(request, responder);
-                }, () -> manageFailureCount(session, request, responder, HumanReadableText.AUTHENTICATION_FAILED));
+                .ifPresentOrElse(username -> doOAuthSuccess(username, session, request, responder),
+                    () -> manageFailureCount(session, request, responder, HumanReadableText.AUTHENTICATION_FAILED));
         }
         session.stopDetectingCommandInjection();
     }
 
+    private void doOAuthSuccess(Username username, ImapSession session, ImapRequest request, Responder responder) {
+        session.authenticated();
+        session.setMailboxSession(getMailboxManager().createSystemSession(username));
+        okComplete(request, responder);
+    }
```

Comment:
Nothing related to Oauth.

Could be reused for other authentication schemes? Could be named `authSuccess` ?